### PR TITLE
Issue #2 | Add react-router-dom and links

### LINKED ISSUE #2: 2. As a user, I want to be able to navigate between the “list” view and the “add an item” view so that the interface is clearly delinteated between viewing and adding.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.

AC:

* `react-router-dom` has been added as a project dependency
* Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
* When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
* Whichever view is selected, the corresponding link should display in bold text

### PR DESCRIPTION
**Pull request by deboragaleano**
## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓  | :sparkles: New feature     |
|  ✓ | :link: Update dependencies |

## Acceptance Criteria

- [x] `react-router-dom` has been added as a project dependency

- [x] Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view

- [x] When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed

- [x] Whichever view is selected, the corresponding link should display in bold text

## Description

* `react-router-dom` was added as a project dependency in order to create 2 links: "List" and "Add item"
*  3 new components were created to satisfy this functionality: `List`, `AddItem` and `Navigation` 
*  **Outcome**: the user is now able to navigate between the “list” view and the “add an item” view. The interface is clearly delineated between viewing the list content and adding a new item.

## Related Issue

Closes #2 

## Testing Steps / QA Criteria

* From your terminal, pull down this branch with git pull origin `ky-dg-react-router-dom` and check that branch out with git checkout `ky-dg-react-router-dom`
* Then `yarn` to install the newly added dependencies locally and `yarn start` to launch the app.
* Click on "List" (left) and "Add Item" (right) links at the very bottom of the page. Make sure that when the links are clicked, the browser URL updates to represent each link's component. Also, the selected link should be displayed in **bold**.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-      </header>
+      <BrowserRouter>
+        <div>
+          <Switch>
+            <Route exact path="/" component={List} />
+            <Route exact path="/add" component={AddItem} />
+          </Switch>
+        </div>
+        <div className="navbar">
+          <Navigation />
```
Comment (hajarNasr) [inline src/App.js:21]: I like that you used a Navigation component with NavLinks and everything here. 👍

File: src/components/Navigation.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
```
Comment (hajarNasr) [inline src/components/Navigation.js:2]: 👍

File: src/components/Navigation.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import './main.css';
+
+const Navigation = () => {
+  return (
+    <div className="navlinks">
+      <NavLink activeClassName="selected" className="list" to="/" exact>
```
Comment (saraheisa) [inline src/components/Navigation.js:8]: Good choice🙌 going with navlink over link as it comes with some cool additional styling attributes like the one you're using here _activeClassName_

**Comment by deboragaleano** [comment]
Just requesting feedback from our mentors, when you get a chance 😃  
Thank you so much! 👍  
@llunaplanet @Lukabaramishvili and Phoebe but I couldn't find her here somehow!

**Review by llunaplanet — APPROVED** [review]
Nice work! Looks good to me!

File: src/components/main.css [hunk 4]
```diff
@@ -0,0 +1,21 @@
+.navlinks {
```
Comment (llunaplanet) [inline src/components/main.css:1]: There is something quite subtle but that helps a lot in the communication as a team, it's agreeing on a code style, for example in this css file you have used both **CamelCase** notation and **lowercase** in the names, **AddItem**, **selected**, etc And for example @hajarNasr and @saraheisa have choosed lowercase with dashes ej: **add-item**. This comment is non blocking, you don't need to change anything, I only wanted to trigger some reflection

Comment (deboragaleano) [reply src/components/main.css:1]: Yes, good point! Thank you for pointing it out! 
I'll bring it up to the group so we can agree on a code style, specially for the CSS file 👍

**Pull request merged**